### PR TITLE
Export geoagents and raster cells

### PR DESCRIPTION
**Pull request by wang-boyu**
- Extract raster cells as numpy array and save to file. Addresses https://github.com/projectmesa/mesa-geo/issues/91#issuecomment-1196219970.
- Extract geoagents as geodataframe.
- Update rainfall and geo_schelling examples to demonstrate.

### CODE REVIEW

File: mesa_geo/raster_layers.py [hunk 1]
```diff
@@ -415,6 +460,28 @@ def from_file(
...
+            "w",
+            driver=driver,
+            width=self.width,
+            height=self.height,
+            count=data.shape[0],
+            dtype=data.dtype,
+            transform=self.transform,
+        ) as dataset:
+            dataset.write(data)
+            dataset.crs = self.crs
```
Comment (rht) [inline mesa_geo/raster_layers.py:482]: What is the usage of the following 2 lines?

Comment (wang-boyu) [reply mesa_geo/raster_layers.py]: Those 2 lines are a mistake... It is now fixed.

File: mesa_geo/raster_layers.py [hunk 2]
```diff
@@ -234,13 +240,52 @@ def coord_iter(self) -> Iterator[Tuple[Cell, int, int]]:
...
+                f"Expected {(1, self.height, self.width)}, received {data.shape}."
+            )
         if attr_name is None:
             attr_name = f"attribute_{len(self.cell_cls.__dict__)}"
+        self._attributes.add(attr_name)
         for x in range(self.width):
             for y in range(self.height):
                 setattr(self.cells[x][y], attr_name, data[0, self.height - y - 1, x])
 
+    def get_raster(self, attr_name: str = None) -> np.ndarray:
```
Comment (rht) [inline mesa_geo/raster_layers.py:264]: The annotation should be `str | None`?

File: mesa_geo/raster_layers.py [hunk 3]
```diff
@@ -415,6 +460,27 @@ def from_file(
             obj.apply_raster(values, attr_name=attr_name)
             return obj
 
+    def to_file(self, raster_file, attr_name: str = None, driver="GTiff"):
```
Comment (rht) [inline mesa_geo/raster_layers.py:463]: Same here. Also, `raster_file` is not yet annotated.

File: mesa_geo/raster_layers.py (hunk 2 above)
Comment (wang-boyu) [reply mesa_geo/raster_layers.py]: Fixed!

File: mesa_geo/raster_layers.py (hunk 3 above)
Comment (wang-boyu) [reply mesa_geo/raster_layers.py]: Updated too.

**Pull request merged**